public/dialog.html

```diff
@@ -11,38 +11,29 @@
 	<link href="https://fonts.googleapis.com/css2?family=Monoton&display=swap" rel="stylesheet">
 	<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
 	<script>var exports = {};</script>
-	<script src='Ad4mConnectDialog.js'></script>
 		
 </head>
 
 <body>
 	<div id="container">
 	</div>
-</body>
+	<script type="module">
+		const { ipcRenderer } = require("electron");
 
-<script>
-	const { ipcRenderer } = require("electron");
-	function run() {
+		import Ad4mConnect from "../dist/web.js";
 		const { appName, appIconPath, executorUrl, capabilityToken, capabilities } = ipcRenderer.sendSync('get');
-		console.log({ appName, appIconPath, executorUrl, capabilityToken, capabilities })
-		const tempTarget = document.createElement('div')
-		const dialog = new Ad4mConnectDialog({ target: tempTarget });
-		dialog.appName = appName;
-		dialog.appIconPath = appIconPath;
-		dialog.executorUrl = executorUrl;
-		dialog.capToken = capabilityToken;
-		dialog.capabilities = capabilities
-		//dialog.showQrScanner = true
-		dialog.resolve = (executorUrl, capabilityToken, client) => {
+		console.log('capabilities', capabilities)
+		const ele = document.createElement('ad4m-connect');
+		ele.appname = appName;
+		ele.appiconpath = appIconPath;
+		ele.url = executorUrl;
+		ele.token = capabilityToken;
+		ele.capabilities = JSON.stringify(capabilities)
+		document.body.appendChild(ele);
+		const client = ele.getClient();
+		client.addEventListener('connected_with_capabilities', (jwt) => {
```

Comment:
We could now switch this to also the the markup initialization, right?

Reply:
yes